library/kani_macros/src/sysroot/contracts/shared.rs

```diff
@@ -144,3 +147,151 @@ pub fn try_as_result_assign(stmt: &syn::Stmt) -> Option<&syn::LocalInit> {
 pub fn try_as_result_assign_mut(stmt: &mut syn::Stmt) -> Option<&mut syn::LocalInit> {
     try_as_result_assign_pat!(stmt, as_mut)
 }
+
+pub fn count_remembers(stmt_vec: &Vec<syn::Stmt>) -> usize {
+    stmt_vec
+        .iter()
+        .filter(|&s: &&syn::Stmt| match s {
+            Stmt::Local(Local { attrs: _, let_token: _, pat, init: _, semi_token: _ }) => match pat
+            {
+                syn::Pat::Ident(PatIdent {
+                    attrs: _,
+                    by_ref: _,
+                    mutability: _,
+                    ident,
+                    subpat: _,
+                }) => ident.to_string().starts_with("remember_kani_internal_"),
+                _ => false,
+            },
+            _ => false,
+        })
+        .count()
+}
+
+pub fn build_ensures(data: &ExprClosure, remember_count: usize) -> (TokenStream2, Expr) {
```

Comment:
please doc what this is doing... I find that giving an example of the expansion is very helpful.

Reply:
I now added various documentation in shared.rs and library/kani/src/contracts.rs, and added an example of the expansion within mod.rs